### PR TITLE
Book Tracking Popover

### PR DESCRIPTION
**Pull request by granttebeau**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/866744728/pulses/1422036969)

Added the book tracking for the admin school directory page. Clicking the "Books" button opens up a popup menu that allows the admin user to view/add/edit/delete book logs

## This PR

Added a "BookLogsMenu" component that displays the popup, and an "EditBookLogs" component that displays the edit functionality. The logic for these components is handled in the "index.tsx" file in the schoolDirectory container. This is where each of the CRUD functions are handled.

I had to change some of the types on the book logs. For tracking purposes, I added an id to the BookLogRequest type. It now extends a "BookLogPostRequest" type that contains the data needed for the create book log endpoint. Also, since the AntD DatePicker needs the date to be an instance of Moment, I installed moment and updated this "date" field type where it needed to be. 

## Screenshots

 

Screen Shot 2021-07-12 at 6 03 03 PM

Screen Shot 2021-07-12 at 6 03 12 PM

## Verification Steps

 

npm start and log in with an admin account. Then, go to the school directory page and click on one of the dropdown menus, and then click "Books" and the popup will open up

### CODE REVIEW

**Review by rymaju — CHANGES_REQUESTED** [review]
Dont install moment, its a peer dependency so you can already import it

**Comment by granttebeau** [comment]
@rymaju fixed, I honestly didn't know you could do that lol

**Review by rymaju — APPROVED** [review]
lgtm

File: src/components/schoolDirectory/BookLogsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+                Edit
+              </Button>
+            </div>
+          ),
+        };
+      },
+    },
+  ];
+
+  switch (currentBookLogs.kind) {
```
Comment (wwohlers) [inline src/components/schoolDirectory/BookLogsMenu.tsx:175]: Not super important here, but in the future you should break this up into multiple components. Try to keep components under 150 lines

File: src/components/schoolDirectory/BookLogsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+                  pagination={{ pageSize: 7 }}
+                />
+              </FormPiece>
+            </Col>
+          </Row>
+          <Footer>
+            <Row gutter={[0, 24]}>
+              <Col flex={24}>
+                <CancelButton
+                  onClick={() => {
```
Comment (wwohlers) [inline src/components/schoolDirectory/BookLogsMenu.tsx:250]: `onClick={onCancel}`

File: src/components/schoolDirectory/BookLogsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                    onCancel();
+                  }}
+                >
+                  Cancel
+                </CancelButton>
+                <SaveButton
+                  onClick={() => {
```
Comment (wwohlers) [inline src/components/schoolDirectory/BookLogsMenu.tsx:257]: `onClick={onSave}`

File: src/components/schoolDirectory/EditBookLog.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                <FormTextArea placeholder="Notes" />
+              </Form.Item>
+            </FormPiece>
+          </Col>
+        </Row>
+        <Footer>
+          <Row gutter={[0, 24]}>
+            <Col flex={24}>
+              <CancelButton
+                onClick={() => {
```
Comment (wwohlers) [inline src/components/schoolDirectory/EditBookLog.tsx:66]: Same change as above

**Review by wwohlers — CHANGES_REQUESTED** [review]
Some small changes, otherwise looks good

**Comment by granttebeau** [comment]
@wwohlers changes have been made

File: src/components/schoolDirectory/BookLogsMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+}) => {
+  const currentBookLogs: AsyncRequest<BookLogRequest[], any> = useSelector(
+    (state: C4CState) => state.bookLogsState.bookLogs,
+  );
+
+  const columns: ColumnType<BookLogWithStyling>[] = [
+    {
+      title: 'Book Amount',
+      dataIndex: 'count',
+      key: 'id',
```
Comment (maxsebso) [inline src/components/schoolDirectory/BookLogsMenu.tsx:94]: key is not needed as we have a dataIndex

**Review by maxsebso — CHANGES_REQUESTED** [review]
Just remove the key field and then the rest looks good!

**Review by justinkonecny — APPROVED** [review]

**Pull request merged**